Using rocm to compile kokkos has an undefined error related to meam

**Summary**

I use rocm to compile on a machine with an AMD graphics card, and I encountered an undefined error when generating the executable file of lammps.
`[100%] Linking CXX executable lmp_mpi`
`liblammps_mpi.so.0: undefined reference to 'LAMMPS_NS::PairMEAMMSKokkos<Kokkos::Serial>::PairMEAMMSKokkos(LAMMPS_NS::LAMMPS*)'`
`liblammps_mpi.so.0: undefined reference to 'LAMMPS_NS::PairMEAMKokkos<Kokkos::Serial>::PairMEAMKokkos(LAMMPS_NS::LAMMPS*)'`
`clang-14: error: linker command failed with exit code 1 (use -v to see invocation)`

**LAMMPS Version and Platform**
Lammps version is lammps-15Jun2023
Linux mymachine 3.10.0-957.nfs.5.x86_64
Gnu compiler is devtoolset_7.3.1
The rocm version is 4.0.1, but I think other versions should be able to reproduce this problem

**Further Information, Files, and Links**
The cmake command is
`
cmake \
-C ../cmake/presets/hip.cmake \
-C ../cmake/presets/most.cmake \
-C ../cmake/presets/hip_amd.cmake \
-DLAMMPS_MACHINE=mpi \
-DBUILD_SHARED_LIBS=on \
-DBUILD_MPI=yes \
-DBUILD_TOOLS=yes \
-DFFT=FFTW3 \
-DFFTW3_LIBRARY=/public/software/mathlib/fftw/gnu/lib/libfftw3.so \
-DFFTW3_INCLUDE_DIR=/public/software/fftw/gnu/include \
-DCMAKE_CXX_COMPILER=hipcc \
-DKokkos_ENABLE_HIP=yes \
-DKokkos_ARCH_VEGA906=yes \
-DPKG_KOKKOS=yes \
-DKokkos_ARCH_VEGA906=yes \
-DHIP_ARCH=gfx906 \
-DPKG_KSPACE=ON \
-DLAMMPS_INSTALL_RPATH=ON \
-DKokkos_CXX_STANDARD=17 \
-DKokkos_ENABLE_HWLOC=ON \
-DKokkos_HWLOC_DIR=/opt/hpc/software/mpi/hwloc \
../cmake/ \
-DMPI_CXX_COMPILER=mpic++ \
-DBUILD_OMP=yes \
-DCMAKE_INSTALL_PREFIX=/work/software/lammps_kokkos/install
`
I found that the rocm-related files may lack some definitions, which caused problems in my execution program generation.Is there any recent plan to fix this problem?

Thanks.

--- a/src/KOKKOS/pair_meam_kokkos.cpp
+++ b/src/KOKKOS/pair_meam_kokkos.cpp
@@ -980,7 +980,7 @@ void PairMEAMKokkos<DeviceType>::update_meam_views()
 
 namespace LAMMPS_NS {
 template class PairMEAMKokkos<LMPDeviceType>;
-#ifdef KOKKOS_ENABLE_CUDA
+#ifdef LMP_KOKKOS_GPU
 template class PairMEAMKokkos<LMPHostType>;
 #endif
 }

--- a/src/KOKKOS/pair_meam_ms_kokkos.cpp
+++ b/src/KOKKOS/pair_meam_ms_kokkos.cpp
@@ -26,7 +26,7 @@ PairMEAMMSKokkos<DeviceType>::PairMEAMMSKokkos(LAMMPS *lmp) : PairMEAMKokkos<Dev
 
 namespace LAMMPS_NS {
 template class PairMEAMMSKokkos<LMPDeviceType>;
-#ifdef KOKKOS_ENABLE_CUDA
+#ifdef LMP_KOKKOS_GPU
 template class PairMEAMMSKokkos<LMPHostType>;
 #endif
 }

--- a/src/KOKKOS/pppm_kokkos.h
+++ b/src/KOKKOS/pppm_kokkos.h
@@ -33,7 +33,7 @@ KSpaceStyle(pppm/kk/host,PPPMKokkos<LMPHostType>);
 
 // clang-format off
 
-// fix up FFT defines for KOKKOS with CUDA
+// fix up FFT defines for KOKKOS with CUDA and HIP
 
 #ifdef KOKKOS_ENABLE_CUDA
 # if defined(FFT_FFTW)
@@ -46,6 +46,19 @@ KSpaceStyle(pppm/kk/host,PPPMKokkos<LMPHostType>);
 #  undef FFT_MKL
 # endif
 # if !defined(FFT_CUFFT) && !defined(FFT_KISSFFT)
+#  define FFT_KISSFFT
+# endif
+#elif defined(KOKKOS_ENABLE_HIP)
+# if defined(FFT_FFTW)
+#  undef FFT_FFTW
+# endif
+# if defined(FFT_FFTW3)
+#  undef FFT_FFTW3
+# endif
+# if defined(FFT_MKL)
+#  undef FFT_MKL
+# endif
+# if !defined(FFT_HIPFFT) && !defined(FFT_KISSFFT)
 #  define FFT_KISSFFT
 # endif
 #endif

